srnca/utils.py

```diff
@@ -87,31 +88,37 @@ def read_image(url, max_size=None):
     
     img = sio.imread(url)
     
+    dim_x, dim_y = img.shape[0], img.shape[1]
+
+    min_dim = min([dim_x, dim_y])
+    img = img[:min_dim, :min_dim]
+
     if max_size is not None:
         img = skimage.transform.resize(img, (max_size, max_size))
-    
    
     img = np.float32(img)/ img.max()
     
     return img
 
-def image_to_tensor(img):
+def image_to_tensor(img, device="cpu"):
 
     if len(img.shape) == 2:
-        my_tensor = torch.tensor(img[np.newaxis, np.newaxis, ...])
+        my_tensor = torch.tensor(img[np.newaxis, np.newaxis, ...], \
+                device=torch.device(device))
     elif len(img.shape) == 3:
-        my_tensor = torch.tensor(img.transpose(2,0,1)[np.newaxis,...])
+        my_tensor = torch.tensor(img.transpose(2,0,1)[np.newaxis,...], \
+                device=torch.device(device))
     
-    return my_tensor
+    return my_tensor#.to(torch.device(device))
 
 def tensor_to_image(my_tensor, index=0):
 
     if my_tensor.shape[1] == 1:
         # rgb or rgba images, convert to rgb
-        img = my_tensor[index,0,:,:]
+        img = my_tensor[index,0,:,:].detach().cpu().numpy()
     else:
         # monochrome images
-        img = my_tensor[index,:3,:,:].permute(1,2,0).detach().numpy()
+        img = my_tensor[index,:3,:,:].permute(1,2,0).detach().cpu().numpy()
```

Comment:
Images need to go to numpy, therefore need to be detached from grad tracking and sent to the cpu.